x-pack/plugins/cases/server/client/metrics/all_cases/aggregations/status_counts.test.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CasePersistedStatus } from '../../../../common/types/case';
+import { StatusCounts } from './status_counts';
+
+describe('StatusCounts', () => {
+  it('returns the correct aggregation', async () => {
+    const agg = new StatusCounts();
+
+    expect(agg.build()).toEqual({
+      status: {
+        terms: {
+          field: 'cases.attributes.status',
+          size: 3,
+        },
+      },
+    });
+  });
+
+  it('formats the response correctly', async () => {
+    const agg = new StatusCounts();
+    const res = agg.formatResponse({
+      status: {
+        buckets: [
+          { key: CasePersistedStatus.OPEN, doc_count: 2 },
+          { key: CasePersistedStatus.IN_PROGRESS, doc_count: 1 },
+          { key: CasePersistedStatus.CLOSED, doc_count: 3 },
+        ],
+      },
+    });
+    expect(res).toEqual({ status: { open: 2, inProgress: 1, closed: 3 } });
+  });
+
+  it('formats the response correctly if the res is undefined', async () => {
+    const agg = new StatusCounts();
+    // @ts-expect-error
+    const res = agg.formatResponse();
```

Comment:
do we somewhere acknowledge that something weird is going on? It's not the same to track 0 because there is no open cases than tracking 0 because there was an error

Reply:
Tbh I followed the patterns we have in the other metrics. Not sure what could happen and the response could be `undefined`. We wanted to ensure the metrics would not fail on weird responses.